### PR TITLE
Tweaks to remote index docs

### PR DESCRIPTION
**Pull request by sam-mccall**
- reduce duplication, particularly between llvm-remote-index and remote-index
- add introduction to remote-index, describing what the feature does
- describe three components of remote-index rather than two (client!)
- clarify support of testing instance (unsupported, not best-effort)
- more direct/concise language (e.g. indexing pipeline -> indexer)
- move a few paragraphs to more closely related sections
- make privacy section of llvm-remote-index specific and remove some subjective language
- broken links fixed (link to *.html, not *.md)

### CODE REVIEW

**Comment by sam-mccall** [comment]
Sorry, I was distracted and missed the review for these docs. I hope it's not too annoying to have these suggestions arrive late. Happy to drop some if you prefer to pick and choose.

Main focus was: 
 - technical cleanups: grammar, simplifying sentences
 - a bit more overview/context at the expense of a some of the detail/duplication

LMK what you think!

File: remote-index.md [hunk 1]
```diff
@@ -1,66 +1,69 @@
...
-## Indexing pipeline
+## Indexer
 
-Indexing pipeline uses `clangd-indexer` under the hood and collects symbols
-that can be exported from headers (functions, classes, types, etc). Running
-`clangd-indexer` is expensive and produced index is not incremental, so using
-the remote index most likely results in a stale index that is periodically
-updated on the server.
+`clangd-indexer` collects symbols that can be exported from headers
+(functions, classes, types, etc).
```
Comment (kirillbobyrev) [inline remote-index.md:24]: Is this supposed to be a separate paragraph or just the end of a sentence? The newline here (way below the usual widthlimit) is a bit misleading.

File: remote-index.md [hunk 2]
```diff
@@ -1,66 +1,69 @@
...
+for large projects on slower machines like laptops. A "remote index" allows
+a shared server to host the index instead.
+
 ## How it works
 
-Remote index works in two separate components, an indexing pipeline and a
-serving infrastructure. The former produces monolithic index file for a
-project of interest and the latter serves information from a monolithic index
-file to clangd clients using a [gRPC](https://grpc.io/)-based server.
+The remote index has three components:
```
Comment (kirillbobyrev) [inline remote-index.md:9]: Markdown likes newlines before and after the lists

**Review by kirillbobyrev — APPROVED** [review]
Sure thing, this does indeed look like an improvement over pushed docs. Thanks for doing this!

No major concerns, just a couple of tiny nits.

**Pull request merged**